src/Microsoft.VisualStudio.Threading.Tests/AsyncReaderWriterLockTests.cs

```diff
@@ -2588,9 +2588,58 @@ public void PrecancelledWriteLockAsyncRequestOnSTA()
             await readlockTask.WithTimeout(UnexpectedTimeout);
         }
 
-#endregion
+        [Fact]
+        public async Task CancelWriteLockUnblocksReadLocks()
```

Comment:
Should we have a test for upgradeable read locks getting unblocked?
Should we add a test where multiple write locks are pending, to verify that when one write lock is canceled, it doesn't permit read locks to be issued if there is still an outstanding write lock waiting?

Reply:
Good suggestions.  I added two more unit tests